Widget fails to initialize when `target` is a SkyCoord

bmorris3 opened the pull request:
On ipyaladin dev, a user can specify a target with a `SkyCoord` like so:
```python
from astropy.coordinates import SkyCoord
from ipyaladin import Aladin

coord = SkyCoord.from_name('M31')

# works
a = Aladin()
a.target = coord
```
but if you initialize the object with a `SkyCoord` target like this:
```python
# doesn't work
a = Aladin(
    target=coord
)
```
it will fail, though the docs say SkyCoords are supported.

The error occurs when the `__init__` method tries to update the `_init_options` traitlet with `SkyCoord` target entries, because `SkyCoord` objects are not json serializable:
https://github.com/cds-astro/ipyaladin/blob/c5fb55f4f5ff519cdd5ca4e0c743338ea7adb160/src/ipyaladin/widget.py#L234-L236

In this PR I've updated `__init__` to remove any `target` entries from `init_options` before passing them into the `_init_options` traitlet. This repairs the `__init__` support for `SkyCoord` targets. 

I haven't seen `_init_options` used elsewhere, so I don't expect this change will break anything. Please correct me if I'm wrong.

ManonMarchand commented:
Hi Brett, 

Thanks for your contribution!

This also happens when setting `fov` with an astropy `Angle`. Could you pop it too?

Also, this means that there is a blind spot in our tests. Could you edit the first cell of `02_Base_commands.ipynb` so that the view is initialized with a SkyCoord object as a target and an Angle as the fov ? (You can squeeze your favorite object there, it's edited in the next cells anyhow)

I don't think we need a changelog entry for this as the bug was never shipped in production.

PS: ipyaladin is not yet on top of my pile of things to do (coming back is tough), but I might get to it next week!

bmorris3 commented:
I'm on it, thanks @ManonMarchand. 

Speaking of rotation, the constructor expects the kwarg `north_pole_orientation`, but the rest of the code uses the `rotation` traitlet. So you can't set the initial rotation with `Aladin(rotation=10 * u.deg)`. One way to fix this:
1. constructor expects `rotation` as the kwarg, update the docs to match
2. deprecate `north_pole_orientation` in the constructor and warn the user to use `rotation` instead.

Would it be acceptable to add those updates to this PR?

bmorris3 commented:
I've gone ahead and made these updates: 

1) Use the same fix for `fov` and `rotation`

2) Since the `pop` fix would remove information from the `_init_options` traitlet, but we're casting each of these attrs to json serializable types anyway, I've added the pure python translations of the objects into `_init_options`. Now calling `aladin._init_options` gives:
```
{'show_simbad_pointer_control': True,
 'show_coo_grid_control': True,
 'show_settings_control': True,
 'show_context_menu': True,
 'coo_frame': 'ICRSd',
 'fov': 10.0,
 'rotation': 90.0,
 'target': '10.6847083 41.26875'}
```

2) Deprecated `north_pole_orientation` in favor of `rotation`.

3) Updated the deprecation warnings throughout `widget.py` which said
> This will be removed in version 1.0.0 (coming after 0.5).

by removing the "(coming after 0.5)" part, since we're on v0.6.0 now.


The following now runs correctly, and stores the initial values as intended:
```python
app = Aladin(
    target=SkyCoord(0, 0, units=u.deg),
    fov=10 * u.deg,
    rotation=90 * u.deg,
)
```

ManonMarchand commented:
Thanks a lot for this!
